core/src/main/java/org/apache/iceberg/BaseMetastoreTableOperations.java

```diff
@@ -43,17 +41,15 @@
   private static final String METADATA_FOLDER_NAME = "metadata";
   private static final String DATA_FOLDER_NAME = "data";
 
-  private final Configuration conf;
   private final FileIO fileIo;
 
   private TableMetadata currentMetadata = null;
   private String currentMetadataLocation = null;
   private boolean shouldRefresh = true;
   private int version = -1;
 
-  protected BaseMetastoreTableOperations(Configuration conf) {
-    this.conf = conf;
-    this.fileIo = new HadoopFileIO(conf);
+  protected BaseMetastoreTableOperations(FileIO fileIo) {
```

Comment:
As an alternative, we can consider providing an abstract method that subclasses should implement. That way, we have more flexibility over initializing `FileIO`. Calling parent constructors must always be the first call in child constructors, which means we will have problems when initializing `FileIO` requires calling another method. 

@jerryshao  what do you think?

Reply:
OK, let me update the code.

Reply:
@aokolnychyi please help to review again, thanks!